emspy/query/profile.py

```diff
@@ -179,25 +181,61 @@ def get_events(self, flight_id, field_type='measurements'):
             selected event field for selected flight
         """
         profile_results = self.__query_profile_results(flight_id)
-        events = pd.DataFrame(profile_results['events']).sort_values('itemId')
-        if field_type == 'measurements':
-            record_type = 'measurement'
-            event_field = 'globalMeasurements'
-        elif field_type == 'timepoints':
-            record_type = 'timepoint'
-            event_field = 'globalTimepoints'
-        glossary = self.__filter_glossary(record_type, 'eventGlobal')
-        event_data = events.iloc[:, :8]\
-            .join(events[event_field].apply(pd.Series))\
-            .melt(id_vars=events.columns[:8])\
-            .drop('variable', axis=1)
-        event_data = event_data.drop('value', axis=1)\
-            .join(event_data['value'].apply(pd.Series))
-        event_data['name'] = \
-            event_data['itemId'].map(glossary['name'].to_dict())
+        events = pd.DataFrame(profile_results['events'])
+
+        if event_specific_fields:
+            # If the user wants event-specific fields, we need to perform some additional manipulation to format those
+            # correctly.
+
+            # Which kind of event-specific fields does the user want to pull (if applicable)?
+            if field_type == 'measurements':
+                record_type = 'measurement'
+                event_field = 'globalMeasurements'
+            elif field_type == 'timepoints':
+                record_type = 'timepoint'
+                event_field = 'globalTimepoints'
+
+            # Filter the glossary for eventGlobal items.
+            glossary = self.__filter_glossary(record_type, 'eventGlobal')
+
+            # Columns that don't come back as arrays (and instead have simple values).
+            simple_columns = ['recordNumber', 'eventType', 'phaseOfFlight', 'severity', 'status', 'falsePositive',
+                              'startTime', 'endTime']
+
+            # Pull out simple values immediately.
+            # Also attach event_field (either globalMeasurement or globalTimepoint) ids and values.
+            event_data = events.loc[:, simple_columns]\
+                .join(events[event_field].apply(pd.Series))\
+                .melt(id_vars=simple_columns)\
+                .drop('variable', axis=1)
+            # Pull out itemId and dataValue from the value column as their own columns.
+            event_data = event_data.drop('value', axis=1)\
+                .join(event_data['value'].apply(pd.Series))
+            # Convert itemId to a string based on this profile's glossary.
+            event_data['itemName'] = \
+                event_data['itemId'].map(glossary['name'].to_dict())
```

Comment:
I found another bug where this whole thing can fail if the Event doesn't have any measurements or timepoints associated with it.